### DISCUSSION TITLE
issue with libssl with 3x version and authentication error with 1.1.1c version when deploying through aws codebuild,

### DISCUSSION THREAD

**Discussion by sandeepgirnala1TR** [openssl/openssl#24778] [Q&A]
<!--
Thank you for your bug report. If this is your first one,
please take the time to read the following lines before posting it.

NOTE:

    If you're asking about how to use OpenSSL, this isn't the right
    forum.  Please see our User Support resources:
    https://github.com/openssl/openssl/blob/master/SUPPORT.md

Please remember to tell us in what OpenSSL version you found the issue.

For build issues:

    If this is a build issue, please include the configuration output
    as well as a log of all errors.  Don't forget to include the exact
    commands you typed.

    With OpenSSL before 1.1.1, the configuration output comes from the
    configuration command.  With OpenSSL 1.1.1 and on, it's the output
    of `perl configdata.pm --dump`

For other issues:

    If it isn't a build issue, example code or commands to reproduce
    the issue is highly appreciated.
    Also, please remember to tell us if you worked with your own
    OpenSSL build or if it is system provided.

Please remember to put ``` lines before and after any commands plus
output and code, like this:

    ```
    $ echo output output output
    output output output
    ```

    ```
i have installed openssl 3x version and deploying my application through terraform by aws codebuild 5x version, then i got no usable version of libssl was found, then i have come up with other soultion by installing 1.1.1c version , then error was gone but it showing some authentication error i have used below configuration to install openssl
- yum groupinstall -y "Development Tools"
  - yum install -y wget
  - wget https://www.openssl.org/source/openssl-1.1.1c.tar.gz
  - tar -zxf openssl-1.1.1c.tar.gz
  - cd openssl-1.1.1c
  - ./config --prefix=/usr/local/openssl --openssldir=/usr/local/openssl
  - make
  - make install
  - echo Updating PATH...
  - export PATH=/usr/local/openssl/bin:$PATH
  - export LD_LIBRARY_PATH=/usr/local/openssl/lib:$LD_LIBRARY_PATH   please provide me the solution and the below is the error
  - <html>
<body>
<!--StartFragment-->
│ Program: /opt/microsoft/powershell/7/pwsh
--
17545 | │ Error Message: Exception:
17546 | │ Line \|
17547 | │    3 \|        throw "FAILED AUTHENTIFICATION:
17548 | │ https://api.github.com/user"
17549 | │      \|
17550 | │ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
17551 | │      \| FAILED AUTHENTIFICATION:
17552 | │ https://api.github.com/user
17553 | │
17554 | │ State: exit status 1

<!--EndFragment-->
</body>
</html>
-->

**Comment by t8m** [comment]
I am sorry but this is not really actionable bug report. The errors in the description are not coming from OpenSSL.

**Comment by rsbeckerca** [comment]
This could be a behaviour change at github.com (from months ago) requiring API tokens instead of sign-on authentication for API usage. @t8m is likely correct that this is not an OpenSSL issue.

**Comment by sandeepgirnala1TR** [comment]
ok but using the 3x version, i got the error for **NO usable version of libssl was found** is the error, what is to be done to remove this error

**Comment by nhorman** [comment]
That's an application level error, so its going to depend on the application you are using.  It looks like you are using a powershell script that runs some application which is dynamically linked to a version of libssl, but its likely specifically looking for libssl.so.1 and you've built and installed libssl.so.3.  You need to rebuild whatever app that your powershell script is calling to link against libssl.so.3
